targets/why2025/badge_bsp_display.c

```diff
@@ -0,0 +1,86 @@
+// Board support package API: WHY2025 implementation
+// SPDX-FileCopyrightText: 2024 Nicolai Electronics
+// SPDX-License-Identifier: MIT
+
+#include <stdbool.h>
+#include <stdint.h>
+#include <string.h>
+#include "bsp/device.h"
+#include "driver/gpio.h"
+#include "dsi_panel_nicolaielectronics_st7701.h"
+#include "esp_err.h"
+#include "esp_lcd_mipi_dsi.h"
+#include "esp_lcd_panel_ops.h"
+#include "esp_ldo_regulator.h"
+#include "esp_log.h"
+#include "freertos/FreeRTOS.h"
+
+static const char* TAG = "BSP display";
+
+static esp_ldo_channel_handle_t ldo_mipi_phy = NULL;
+
+static bool bsp_display_initialized = false;
+
+static esp_err_t bsp_display_enable_dsi_phy_power(void) {
+    esp_ldo_channel_config_t ldo_mipi_phy_config = {
+        .chan_id = 3,
+        .voltage_mv = 2500,
+    };
+    return esp_ldo_acquire_channel(&ldo_mipi_phy_config, &ldo_mipi_phy);
+}
+
+static esp_err_t bsp_display_reset(void) {
+    gpio_config_t lcd_reset_conf = {
+        .pin_bit_mask = BIT64(14),
```

Comment:
Could use some #defines here instead of using raw pin numbers to make it more clear what is going on and make things easier to change later

Reply:
Changed :smile: